projects/batfish-client/test/org/batfish/client/ClientTest.java

```diff
@@ -0,0 +1,461 @@
+      JsonNode comparatorNode = mapper.readTree("\">=\"");
+      Question.InstanceData.Variable variable
+            = new Question.InstanceData.Variable();
+      variable.setType(Question.InstanceData.Variable.Type.COMPARATOR);
+      Client.validateType(comparatorNode, variable);
+   }
+
+   @Test
+   public void testInvalidIntegerValue() throws IOException {
+      String input = "1.5";
+      Question.InstanceData.Variable.Type expectedType
+            = Question.InstanceData.Variable.Type.INTEGER;
+      String expectedMessage =
+            String.format("It is not a valid JSON %s value",
+                  Question.InstanceData.Variable.Type.INTEGER.getName());
+      validateTypeWithInvalidInput(input, expectedMessage, expectedType);
+   }
+
+   @Test
+   public void testValidIntegerValue() throws IOException {
+      JsonNode integerNode = mapper.readTree("15");
+      Question.InstanceData.Variable variable
+            = new Question.InstanceData.Variable();
+      variable.setType(Question.InstanceData.Variable.Type.INTEGER);
+      Client.validateType(integerNode, variable);
+   }
+
+   @Test
+   public void testInvalidIPValue() throws IOException {
+      String input = "\"0.0.0\"";
+      JsonNode inputNode = mapper.readTree(input);
+      Question.InstanceData.Variable.Type expectedType
+            = IP;
+      String expectedMessage = "It is not a valid IP address";
+      validateTypeWithInvalidInput(input, expectedMessage, expectedType);
+   }
+
+   @Test
+   public void testValidIPValue() throws IOException {
+      JsonNode IPNode = mapper.readTree("\"0.0.0.0\"");
+      Question.InstanceData.Variable variable
+            = new Question.InstanceData.Variable();
+      variable.setType(IP);
+      Client.validateType(IPNode, variable);
+   }
+
+   @Test
+   public void testInvalidJavaRegexValue() throws IOException {
+      String invalidJavaRegex = "\"...{\\\\Q8\\\\E}\"";
+      JsonNode inputNode = mapper.readTree(invalidJavaRegex);
+      Question.InstanceData.Variable.Type expectedType
+            = Question.InstanceData.Variable.Type.JAVA_REGEX;
+      String expectedMessage ="It is not a valid Java regular expression";
+      validateTypeWithInvalidInput(invalidJavaRegex, expectedMessage,
+            expectedType);
+   }
+
+   @Test
+   public void testValidJavaRegexValue() throws IOException {
+      JsonNode inputNode = mapper.readTree("\".*\"");
+      Question.InstanceData.Variable variable
+            = new Question.InstanceData.Variable();
+      variable.setType(Question.InstanceData.Variable.Type.JAVA_REGEX);
+      Client.validateType(inputNode, variable);
+   }
+
+   @Test
+   public void testInvalidJsonPathValue() throws IOException {
+      String input = "{\"variable\" : \"I am variable\"}";
+      Question.InstanceData.Variable.Type expectedType
+            = Question.InstanceData.Variable.Type.JSON_PATH;
+      String expectedMessage =
+            String.format("Missing 'path' element of %s",
```

Comment:
do `=` go at the end of the line or the beginning of the next one? Look through Batfish and see what the standard is.

We should be consistent, but here we do one thing at L285 and another at L286.